src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponent.cs

```diff
@@ -61,6 +62,17 @@ public IPrincipal User
         }
 
         /// <summary>
+        /// Gets the <see cref="ClaimsPrincipal"/> for the current user.
+        /// </summary>
+        public ClaimsPrincipal UserClaimsPrincipal
```

Comment:
`public ClaimsPrincipal UserClaimsPrincipal => ViewContext?.HttpContext?.User` for 💯 points!